scrypto/src/engine/scrypto_env.rs

```diff
@@ -449,6 +449,13 @@ impl ClientTransactionRuntimeApi<ClientApiError> for ScryptoEnv {
 
         scrypto_decode(&actor).map_err(ClientApiError::DecodeError)
     }
+
+    fn panic(&mut self, message: String) -> Result<(), ClientApiError> {
+        unsafe {
+            panic(message.as_ptr(), message.len());
+        };
+        Ok(())
```

Comment:
IMO this function should return `!` and we should possibly have a line here with:
```
// After calling panic, the engine should not return control to the WASM, so this will never be run
// But we include this to make the compiler happy
panic!()
```

Reply:
This is because we use the universal `ClientTransactionRuntimeApi`, where we can't return `!` because RE use `Err` for rolling back.

Reply:
Ahh okay, makes sense. I still think we should panic here instead of returning though (maybe with a comment explaining why the return is an Option)

And similarly, I'd suggest replacing the `loop` with a panic in the caller, but it's very minor :).

As a slightly bigger musing - I wonder if we should call the syscall and error `abort_transaction` or just `abort` instead of `panic` - which feels a little more correct abstraction wise? It's a little weird to see a function / syscall called `panic` which doesn't panic aha :D. From the caller perspective, we're actually requesting the transaction aborts.